### PR TITLE
4: GET /api/articles/:article_id

### PR DESCRIPTION
**Pull request by mrjohnsly**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -62,4 +62,77 @@ describe("/api/topics", () => {
...
+				.then(({ body }) => {
+					expect(body.article).toHaveProperty("author");
+					expect(body.article).toHaveProperty("title");
+					expect(body.article).toHaveProperty("article_id");
+					expect(body.article).toHaveProperty("body");
+					expect(body.article).toHaveProperty("topic");
+					expect(body.article).toHaveProperty("created_at");
+					expect(body.article).toHaveProperty("votes");
+				});
+		});
```
Comment (douglashellowell) [inline __tests__/app.test.js:91]: no need for these to be seperate tests!

You can assert it has a property of article and test the contents in the same test if you want!

Additionally try to make your assertions on the properties more specific!
You know the article is meant to be article_id `1`! Look at the test data and make sure it's that specific article being sent back

File: app/models/articles.models.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+		.then((dbResult) => {
+			if (dbResult.rowCount === 0) {
+				return Promise.reject({ code: 404, message: "No article found" });
+			} else {
+				return dbResult.rows[0];
+			}
+		})
+		.catch((error) => {
+			throw error;
+		});
```
Comment (douglashellowell) [inline app/models/articles.models.js:17]: no need to catch the error just to throw it!

you can remove this and the caller (in the controller) will catch it anyway! :D

**Comment by douglashellowell** [comment]
great stuff!

very good thorough tests

ready to merge 💪🏻

File: app/models/articles.models.js (hunk 2 above)
Comment (mrjohnsly) [reply app/models/articles.models.js:17]: Sorry not entirely sure what you mean by this?

**Pull request merged**